src/core/public/chrome/chrome_service.ts

```diff
@@ -175,6 +183,29 @@ export class ChromeService {
           takeUntil(this.stop$)
         ),
 
+      /**
+       * Sets a named badge which can be used by setBadge
+       */
+      addNamedBadge: (name: string, badge: ChromeBadge) => {
+        namedBadges.set(name, badge);
+      },
+
+      /**
+       * Get an observable of the current badge
+       */
+      getBadge$: (): Rx.Observable<ChromeBadge | undefined> =>
+        badge$.pipe(
+          map(badge => (typeof badge === 'string' ? namedBadges.get(badge) : badge)),
```

Comment:
There's potential for us to make `namedBadges` to be an observable itself, so that if `addNamedBadge` is invoked after `setBadge` and `getBadge$` that it will at this time emit the proper badge. As it's written now, we'll just get `undefined` if this sequence of events occurs.